src/classes/ERR_Notifier.cls

```diff
@@ -61,33 +81,59 @@ public class ERR_Notifier {
 
         if (errorSettings.Error_Notifications_On__c == true && errorNotifRecipient != null) {
 
-            //We will post to chatter, if enabled
-            if(errorNotifRecipient instanceof id && errorNotifRecipient.startsWith(NotificationOptions.chatterGroup) 
-            && Schema.SObjectType.User.isFeedEnabled()) {
-                postErrorsToChatter(errorNotifRecipient);
-            //We will send email
+            ERR_Notifier notifierService = new ERR_Notifier();
+
+            Boolean isChatterGroup = errorNotifRecipient instanceof Id
+                    && errorNotifRecipient.startsWith(ERROR_NOTIFICATION_CHATTER_PREFIX);
+
+            if (isChatterGroup && Schema.SObjectType.User.isFeedEnabled()) {
+
+                notifierService.postErrorsToChatter(errorNotifRecipient);
+
             } else {
-                List<String> sendList = new List<String>();
-                if (errorNotifRecipient instanceof id && errorNotifRecipient.startsWith(NotificationOptions.user)) {
-                    List<User> useremaillist = new List<User>();
-                    useremaillist = [SELECT email FROM User WHERE id = :errorNotifRecipient AND isActive = true];
-                    for (User u : useremaillist)
-                        sendList.add(u.email);
-                } else if(errorNotifRecipient == NotificationOptions.sysAdmins) {
-                    sendList.addAll(getSystemAdminEmails());
-                }
 
-                sendEmailNotifications(context, sendList);
+                notifierService.sendEmailNotifications(context,
+                        notifierService.getErrorEmailRecipients(errorNotifRecipient));
+            }
+        }
+    }
+
+    /** @description Private constructor so the class cannot be instantiated outside of this class */
+    @TestVisible
+    private ERR_Notifier() {}
+
+    // ====================== HELPER METHODS FOR ERROR NOTIFICATION SERVICE =========================
+
+    /**********************************************************************************************
+     * @description Using the provided error notification user id (or 'All System Admins') or gets
+     * all system admin email addresses
+     * @param errorNotifRecipient Id or Null
+     * @return List<Email>
+     */
+    private List<String> getErrorEmailRecipients(String errorNotifRecipient) {
+        List<String> recipientsList = new List<String>();
+        Boolean isUserId = errorNotifRecipient instanceof Id
+                && errorNotifRecipient.startsWith(ERROR_NOTIFICATION_USER_PREFIX);
+
+        if (isUserId) {
+            List<User> users = [SELECT Email FROM User WHERE Id = :errorNotifRecipient AND IsActive = TRUE LIMIT 1];
```

Comment:
So, if the selected notification User is not active, send the error emails to all active admins? That's an option, though I think we'd want to add a header comment to the email saying that the specified user is not active. It's a relatively simple change so I'm not sure it warrants a new WI. I think it could be done with a quick check in this method and in the `sendErrorQueueEmailNotification` method.